### DISCUSSION TITLE
How to use PrimeVue with VitePress?

### DISCUSSION THREAD

**Discussion by onurusluca** [vuejs/vitepress#3332] [Q&A]
Is it not possible to use [PrimeVue ](https://primevue.org/) with VitePress?
I am getting this error:
`button.esm.js:127 Uncaught (in promise) TypeError: _ctx.cx is not a function` 
for this simple usage:
```
<script setup>
import Button from "primevue/button"
</script>
<template>
    <Button label="Submit" />
</template>
```

**My setup:**
`.vitepress/theme/index.ts:`
```
// https://vitepress.dev/guide/custom-theme
import { h } from 'vue'
import type { Theme } from 'vitepress'
import DefaultTheme from 'vitepress/theme-without-fonts'
import './style.css'
import './custom.css'

// PrimeVue
import 'primevue/resources/themes/lara-light-teal/theme.css'
import 'primeicons/primeicons.css'
import Button from 'primevue/button'
import InputText from 'primevue/inputtext'
import InputNumber from 'primevue/inputnumber'
import OverlayPanel from 'primevue/overlaypanel'
import InputSwitch from 'primevue/inputswitch'
import Dropdown from 'primevue/dropdown'

export default {
  extends: DefaultTheme,
  Layout: () => {
    return h(DefaultTheme.Layout, null, {
      // https://vitepress.dev/guide/extending-default-theme#layout-slots
    })
  },
  enhanceApp({ app, router, siteData }) {
    app.component('Button', Button)
    app.component('InputText', InputText)
    app.component('InputNumber', InputNumber)
    app.component('OverlayPanel', OverlayPanel)
    app.component('InputSwitch', InputSwitch)
    app.component('Dropdown', Dropdown)
  }
} satisfies Theme

```

**Comment by teacherhogg** [comment]
Did you ever figure this out? I am getting errors on build (not dev): 

"Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension" 

using node v20.11, vitepress v1.0.0-rc.40 and primevue 3.47.1

**Reply by onurusluca** [reply]
Unfortunately, no. I wasn't. I had to go vanilla.

**Comment by TimonPeng** [comment]
https://github.com/primefaces/primevue-examples/tree/main/vite-unstyled-tailwind

**Comment by liub1934** [comment]
> Did you ever figure this out? I am getting errors on build (not dev):
> 
> "Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension"
> 
> using node v20.11, vitepress v1.0.0-rc.40 and primevue 3.47.1

Me too.
https://github.com/primefaces/primevue/issues/5487

**Comment by fjun99** [comment]
I just tested it, and you might need to add "type": "module" in your `package.json` file. 

please refer to vitepress docs:  https://vitepress.dev/guide/getting-started

> NOTE
> VitePress is an ESM-only package. Don't use require() to import it, and make sure your nearest package.json contains "type": "module", or change the file extension of your relevant files like .vitepress/config.js to .mjs/.mts. Refer to [Vite's troubleshooting guide](http://vitejs.dev/guide/troubleshooting.html#this-package-is-esm-only) for more details. Also, inside async CJS contexts, you can use await import('vitepress') instead.
